packages/rpc-types/src/lamports.ts

```diff
@@ -46,18 +49,45 @@ export function lamports(putativeLamports: bigint): LamportsUnsafeBeyond2Pow53Mi
     return putativeLamports;
 }
 
+type ExtractAdditionalProps<T, U> = Omit<T, keyof U>;
+
 export function getLamportsEncoder(): FixedSizeEncoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
     return getMemoizedU64Encoder();
 }
 
+export function getLamportsEncoderFromEncoder<TEncoder extends NumberEncoder>(
+    innerEncoder: TEncoder,
+): Encoder<LamportsUnsafeBeyond2Pow53Minus1> & ExtractAdditionalProps<TEncoder, NumberEncoder> {
```

Comment:
Are we sure that this return type keeps the fixed-size or variable-size properties of the inner encoder? Should add a few typetests just to be safe?

Reply:
Added typetests: https://github.com/solana-labs/solana-web3.js/blob/7ec7cfcec42b3321201bf79e71df85875d0f993f/packages/rpc-types/src/__typetests__/lamports-typetests.ts